src/utils/validateChainId.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Utility function for validating chain IDs
+ */
+
+const isHexString = /^0x[0-9a-fA-F]+$/
+const maxSafeInteger = BigInt(2147483647)
+
+/**
+ * Validates if a chain ID is valid and optionally matches an expected value
+ * 
+ * @param givenChainId The chain ID to validate (must be a hex string with 0x prefix)
+ * @param expectedChainId Optional expected chain ID to compare against
+ * @returns True if the chain ID is valid (and matches expected value if provided), false otherwise
+ */
+export function validateTxChainId(givenChainId: any, expectedChainId?: number): boolean {
+  // First check if the given chain ID is valid
+  if (givenChainId === undefined || givenChainId === null) {
+    return false;
+  }
+
+  // Only accept 0x-prefixed hex strings
+  if (typeof givenChainId !== 'string' || !givenChainId.startsWith('0x')) {
+    return false;
+  }
+
+  // Validate it's a proper hex string
+  if (!isHexString.test(givenChainId)) {
+    return false;
+  }
+
+  let chainIdBigInt: bigint;
+
+  try {
+    // Convert the hex string to BigInt
+    chainIdBigInt = BigInt(givenChainId);
+  } catch (error) {
+    return false; // Conversion error
+  }
+
+  // EIP-155 specifies that chain IDs should be positive integers
+  // 0 is reserved, so we check if it's greater than 0
+  if (chainIdBigInt <= BigInt(0)) {
+    return false;
+  }
+
+  // Arbitrary large number check to avoid absurdly large chain IDs
+  // The largest known chain ID as of now is much smaller than this
+  if (chainIdBigInt > maxSafeInteger) {
+    return false;
+  }
+
+  // If expected chain ID is provided, validate it and compare
+  if (expectedChainId !== undefined) {
```

Comment:
`expectedChainId` is optional param. what if one doesn't pass it? 
Also, I think we shouldn't really be giving option to pass expectedChainId. It should be compared directly w/ ShardeumFlag.CHAIN_ID

Reply:
That requires having to import shardeumFlags / tie into more global state. Having functions seperate the concerns of where the data comes from lends to easier testing